LPC Projects/pwmdemodulator/src/pwmdemodulator.c

```diff
@@ -0,0 +1,300 @@
+		if (RiseTime != 0)
+		{
+			if (CurrentTimer > k_min_pulse && CurrentTimer < k_max_pulse)
+			{
+				//char buf[100];
+				//sprintf(buf, "%d\n", CurrentTimer);
+				//Board_UARTPutSTR(buf);
+								/* Send a simple one time CAN message */
+				PulseWidth_US = CurrentTimer;
+			    int value = (int)map(PulseWidth_US,
+			                       k_offset_rc_in,
+			                       k_scale_rc_in,
+			                       k_offset_stored_angle,
+			                       k_scale_stored_angle);
+				msg_obj.msgobj  = 0;
+				msg_obj.mode_id = 0x000;
+				msg_obj.mask    = 0x0;
+				msg_obj.dlc     = 4;
+				msg_obj.data[0] = value & 0xFF;
+				msg_obj.data[1] = (value >> 8) & 0xFF;
+				msg_obj.data[2] = (value >> 16) & 0xFF;
+				msg_obj.data[3] = (value >> 24) & 0xFF;
+				LPC_CCAN_API->can_transmit(&msg_obj);
+
+			}
+		}
+	}
+	LPC_TIMER16_0->TC = 0;
+	if (Chip_TIMER_CapturePending(LPC_TIMER16_0, 0))
+	{
+		Chip_TIMER_ClearCapture(LPC_TIMER16_0, 0);
+	}
+}
+
+// Brake PWD
+void TIMER16_1_IRQHandler(void)
+{
+	bool CurrentValue = Chip_GPIO_GetPinState(LPC_GPIO, 1, 8);
+	uint16_t CurrentTimer = LPC_TIMER16_1->CR[0];
+	if (CurrentValue)
+	{
+		BrakeRiseTime = CurrentTimer;
+	}
+	else
+	{
+		if (BrakeRiseTime != 0)
+		{
+			brake = CurrentTimer > 1750 ? true : false;
+		}
+	}
+	LPC_TIMER16_1->TC = 0;
+	if (Chip_TIMER_CapturePending(LPC_TIMER16_1, 0))
+	{
+		Chip_TIMER_ClearCapture(LPC_TIMER16_1, 0);
+	}
+}
+
+// Auton PWD
+void TIMER32_0_IRQHandler(void)
+{
+	bool CurrentValue = Chip_GPIO_GetPinState(LPC_GPIO, 1, 5);
+	uint16_t CurrentTimer = LPC_TIMER32_0->CR[0];
+	if (CurrentValue)
+	{
+		AutonRiseTime = CurrentTimer;
+	}
+	else
+	{
+		if (AutonRiseTime != 0)
+		{
+			auton = CurrentTimer > 1750 ? true : false;
+		}
+	}
+	LPC_TIMER32_0->TC = 0;
+	if (Chip_TIMER_CapturePending(LPC_TIMER32_0, 0))
+	{
+		Chip_TIMER_ClearCapture(LPC_TIMER32_0, 0);
+	}
+}
+
+int main(void)
+{
+	uint32_t CanApiClkInitTable[2];
+	/* Publish CAN Callback Functions */
+	CCAN_CALLBACKS_T callbacks = {
+		CAN_rx,
+		CAN_tx,
+		CAN_error,
+		NULL,
+		NULL,
+		NULL,
+		NULL,
+		NULL,
+	};
+
+    SystemCoreClockUpdate();
+    Board_Init();
+    // Set the LED to the state of "On"
+    Board_LED_Set(0, true);
+    baudrateCalculate(TEST_CCAN_BAUD_RATE, CanApiClkInitTable);
+
+    Chip_GPIO_Init(LPC_GPIO);
+    // Input pins for the receiver
+    Chip_GPIO_SetPinDIRInput(LPC_GPIO, 0, 2);
+    Chip_GPIO_SetPinDIRInput(LPC_GPIO, 1, 8);
+    Chip_GPIO_SetPinDIRInput(LPC_GPIO, 1, 5);
+
+    // Uart crap
+	Chip_IOCON_PinMuxSet(LPC_IOCON, IOCON_PIO1_6, (IOCON_FUNC1 | IOCON_MODE_INACT));/* RXD */
+	Chip_IOCON_PinMuxSet(LPC_IOCON, IOCON_PIO1_7, (IOCON_FUNC1 | IOCON_MODE_INACT));/* TXD */
+	Chip_UART_Init(LPC_USART);
+	Chip_UART_SetBaud(LPC_USART, 115200);
+	Chip_UART_ConfigData(LPC_USART, (UART_LCR_WLEN8 | UART_LCR_SBS_1BIT));
+	Chip_UART_SetupFIFOS(LPC_USART, (UART_FCR_FIFO_EN | UART_FCR_TRG_LEV2));
+	Chip_UART_TXEnable(LPC_USART);
+
+	Board_UARTPutSTR("asdfadsfadfasdf\n");
```

Comment:
Is this supposed to be a startup message?

Reply:
yes I will change that